src/schema/v2/types/recentlySoldArtworksConnection.ts

```diff
@@ -0,0 +1,45 @@
+import { ResolverContext } from "types/graphql"
+import { GraphQLBoolean, GraphQLFieldConfig } from "graphql"
+import { pageable } from "relay-cursor-paging"
+import { connectionFromArray } from "graphql-relay"
+import { artworkConnection } from "schema/v2/artwork"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { createPageCursors } from "schema/v2/fields/pagination"
+import { shuffle } from "lodash"
+
+export interface RecentlySoldArtworksConnectionSource {
+  metadata: { recentlySoldArtworkIDs: string[] }
+}
+
+// This is a function so that execution is deferred; otherwise
+//   `artworkConnection.connectionType` errs because `artworkConnection` is undefined.
```

Comment:
yyyyyyyyyeah I pulled my hair on this one for a while. I'm not sure why `artworksConnection` is undefined at immediate execution but not later....but making this a function gets around that.